README.md

```diff
@@ -24,173 +24,230 @@ Docker images are the preferred way to deploy this code in a production environm
+        - [Coverage Reports](#coverage-reports)
+    - [4. Further setup to run locally](#4-further-setup-to-run-locally)
+        - [OpenSearch](#opensearch)
+        - [Database](#database)
+        - [Environment variables](#environment-variables)
+        - [Storage Service](#storage-service)
+        - [Reporting](#reporting)
+        - [Logging](#logging)
+        - [Firebase Cloud Messaging](#logging)
+        - [OpenSearch Analytics (E-Kirjasto, Finland)](#opensearch-analytics-e-kirjasto-finland)
+    - [5. Running the Application](#5-running-the-application)
+    - [6. Installation Issues](#6-installation-issues)
+- [The Admin Interface](#the-admin-interface)
+    - [1. Access](#1-access)
+    - [2. Creating A Library](#2-creating-a-library)
+    - [3. Adding Collections](#3-adding-collections)
+    - [4. Importing OPDS feeds](#4-importing-opds-feeds)
+    - [5. Generating Search Indices](#5-generating-search-indices)
+    - [6. Patron authentication](#6-patron-authentication)
+    - [7. Troubleshooting](#7-troubleshooting)
+    - [8. Sitewide Settings](#8-sitewide-settings)
+- [Scheduled Jobs](#scheduled-jobs)
+    - [1. Job Requirements](#job-requirements)
+        - [hold_notifications](#hold_notifications)
+        - [loan_notifications](#loan_notifications)
+- [Code Style](#code-style)
+    - [1. Pre-Commit Configuration](#1-pre-commit-configuration)
+    - [2. Linters](#2-linters)
+        - [Built in](#built-in)
+        - [Black](#black)
+        - [isort](#isort)
+        - [autoflake](#autoflake)
+- [Localization (i18n, l10n, flask-pybabel, managing translations)](#localization-i18n-l10n-flask-pybabel-managing-translations)
+- [PyInstrument](#pyinstrument)
+    - [Profiling tests suite](#profiling-tests-suite)
+    - [Environment Variables](#environment-variables-1)
+
+## Setup
+
+### 1. Set up and run the application with Docker Compose
+
+To quickly set up a development environment, we include a [docker-compose-dev.yml](./docker-compose-dev.yml)
+file. This docker-compose file, will build the webapp and scripts containers from your local repository, and start
+those containers as well as all the necessary service containers.
 
-Create a virtual environment with Python 3.11.1, run:
+But first, install docker if not yet installed:
 
-```shell
-make venv
+```sh
+brew install --cask docker
 ```
 
-Activate the virtual environment and install dependencies with Poetry:
+Start the Docker app and then check it's running:
 
-```shell
-pyenv activate circ-311
-poetry install
+```sh
+docker ps
 ```
 
-If all went well, you should see the activated virtual environment `(circ-311)` in your shell and all packages
-installed. You can verify this by checking the Python version:
+Add the url for `ADMIN_EKIRJASTO_AUTHENTICATION_URL` in the `docker-compose-dev.yml` file. Then build the containers by
+running the following command:
 
-```shell
-python3 --version
+```sh
+docker compose -f docker-compose-dev.yml up --build -d
```

Comment:
I had to comment out data-api section in docker-compose-dev.yml, otherwise an error would appear: `Error response from daemon: no matching manifest for linux/arm64/v8 in the manifest list entries: no match for platform in manifest: not found`